compiler/data/function-data.h

```diff
@@ -53,9 +62,14 @@ class FunctionData {
   vector<VarPtr> param_ids;
   vector<FunctionPtr> dep;
   std::set<ClassPtr> class_dep;
+  std::set<ClassPtr> exceptions_thrown; // exceptions that can be thrown by this function
   bool tl_common_h_dep = false;
   FunctionPtr function_in_which_lambda_was_created;
 
+  // test_data is is non-null if there are any kphp-test-* attributes attached to this function;
+  // see FunctionTestData documentation for more info
+  FunctionTestData *test_data;
```

Comment:
I thought it would be useful in the future to test attributes like "pure" (if we'll ever infer them from non-builtin funcs).
For example, you could write something like this:
```php
/**
 * @kphp-test-pure
 */
function should_be_pure() { return 10; }
```
It could be useful outside of tests, but I think the cost of adding a new "public" (well-documented) attribute is higher than having some ways to make function properties testing easier.

We could forbid "kphp-test-*" attributes for the user code.

I want some feedback from other colleagues.

CC @eugene536 @unserialize @DrDet

Reply:
Off topic: we already have `@kphp-pure-function` annotation.

Reply:
I believe that `kphp-pure-function` is only for builtin functions and it marks a function as pure.

Anyway, I wanted to give an example where you test an `X` property of the function directly.
I'll try to prepare better examples to make the intentions more clear.

Reply:
Removed  FunctionTestData.